app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -99,4 +121,53 @@ public Retrofit getRetrofit() {
 
         return retrofit;
     }
+
+    /**
+     * Provides {@SecurePreferences} singleton instance to access shared preferences
+     *
+     * @return A single instance of {@SecurePreferences}
+     */
+    public static SharedPreferences getSharedPreferences() {
+        if (sSecurePreferences == null) {
+            synchronized (BaseApplication.class) {
+                if (sSecurePreferences == null) {
+                    sSecurePreferences = new SecurePreferences(getAppContext(), PREF_PASSWORD,
+                            PREF_FILE_NAME);
+                    SecurePreferences.setLoggingEnabled(BuildConfig.DEBUG);
+                }
+            }
+        }
+        return sSecurePreferences;
+    }
+
+    public static Context getAppContext() {
```

Comment:
Isn't it easier to just use `getApplicationContext()` everywhere? Is it possible to remove this method (to keep the amount of code as small as possible)?

Reply:
No it's impossible.
We need static app context reference for getSharedPreferences(), while getApplicationContext() is a non-static mnethod.